rcl/test/rcl/test_logging_rosout.cpp

```diff
@@ -119,6 +123,10 @@ class TestLoggingRosout : public ::testing::Test
     rcl_ret_t ret = rcl_subscription_fini(this->subscription_ptr, this->node_ptr);
     delete this->subscription_ptr;
     EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    if (rcl_logging_rosout_enabled()) {
```

Comment:
It seems the check condition is weak, `rcl_logging_rosout_enabled()` with true doesn't mean the rosout publisher is initialized. I suggest using

```suggestion
    if (rcl_logging_rosout_enabled() && node_options.enable_rosout) {
```

Reply:
addressed with https://github.com/ros2/rcl/pull/1065/commits/3cd4966626fc71f1b983c2ae811b50f44e857121